core/Pos/Binary/Crypto.hs

```diff
@@ -212,6 +212,7 @@ instance Bi PassPhrase where
     decode = do
         bs <- decode @ByteString
         let bl = BS.length bs
+        -- This check is simple so it doesn't use noCheck.
```

Comment:
Rightly so. This check is actually a legitimate part of deserialization. The form of a serialized passphrase is always 0 or 32 bytes, so we *must* check it in the decoder instance.